src/app/3d/qgspointcloudlayer3drendererwidget.cpp

```diff
@@ -0,0 +1,134 @@
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgspointcloudlayer3drendererwidget.h"
+
+#include "qgspointcloud3dsymbolwidget.h"
+#include "qgspointcloudlayer3drenderer.h"
+
+#include "qgspointcloudlayer.h"
+#include "qgspointcloud3dsymbol.h"
+
+#include <QBoxLayout>
+#include <QCheckBox>
+
+QgsPointCloudLayer3DRendererWidget::QgsPointCloudLayer3DRendererWidget( QgsPointCloudLayer *layer, QgsMapCanvas *canvas, QWidget *parent )
+  : QgsMapLayerConfigWidget( layer, canvas, parent )
+{
+  setPanelTitle( tr( "3D View" ) );
+
+  QVBoxLayout *layout = new QVBoxLayout( this );
+  layout->setContentsMargins( 0, 0, 0, 0 );
+
+  mChkEnabled = new QCheckBox( tr( "Enable 3D Renderer" ), this );
+  layout->addWidget( mChkEnabled );
+
+  mWidgetPointCloudSymbol = new QgsPointCloud3DSymbolWidget( layer, this );
+  layout->addWidget( mWidgetPointCloudSymbol );
+
+  connect( mChkEnabled, &QCheckBox::clicked, this, &QgsPointCloudLayer3DRendererWidget::onEnabledClicked );
+  connect( mWidgetPointCloudSymbol, &QgsPointCloud3DSymbolWidget::changed, this, &QgsPointCloudLayer3DRendererWidget::widgetChanged );
+}
+
+void QgsPointCloudLayer3DRendererWidget::setRenderer( const QgsPointCloudLayer3DRenderer *renderer )
+{
+  mRenderer.reset( renderer ? renderer->clone() : nullptr );
+  if ( renderer != nullptr )
+    mWidgetPointCloudSymbol->setSymbol( const_cast<QgsPointCloud3DSymbol *>( renderer->symbol() ) );
+  whileBlocking( mChkEnabled )->setChecked( renderer ? renderer->symbol()->isEnabled() : false );
+}
+
+QgsPointCloudLayer3DRenderer *QgsPointCloudLayer3DRendererWidget::renderer()
+{
+  // TODO: use unique_ptr for point cloud symbol
+  QgsPointCloud3DSymbol *sym = mWidgetPointCloudSymbol->symbol();
+  mRenderer.reset( new QgsPointCloudLayer3DRenderer );
+  mRenderer->setSymbol( sym );
+  delete sym;
+  mRenderer->setLayer( qobject_cast<QgsPointCloudLayer *>( mLayer ) );
+  return mRenderer.get();
+}
+
+void QgsPointCloudLayer3DRendererWidget::apply()
+{
+  QgsPointCloudLayer3DRenderer *r = nullptr;
+  if ( mChkEnabled->isChecked() )
+  {
+    r = renderer()->clone();
+    if ( r )
+      r->setSymbol( mWidgetPointCloudSymbol->symbol() );
+  }
+  mLayer->setRenderer3D( r );
+}
+
+void QgsPointCloudLayer3DRendererWidget::onEnabledClicked()
+{
+  mWidgetPointCloudSymbol->setEnabled( mChkEnabled->isChecked() );
+  emit widgetChanged();
+}
+
+void QgsPointCloudLayer3DRendererWidget::syncToLayer( QgsMapLayer *layer )
+{
+  mLayer = layer ;
+  QgsPointCloudLayer *pointCloudLayer = qobject_cast<QgsPointCloudLayer *>( layer );
+  mWidgetPointCloudSymbol->setLayer( pointCloudLayer );
+  QgsAbstract3DRenderer *r = layer->renderer3D();
+  if ( r && r->type() == QLatin1String( "pointcloud" ) )
+  {
+    QgsPointCloudLayer3DRenderer *pointCloudRenderer = static_cast<QgsPointCloudLayer3DRenderer *>( r );
```

Comment:
This isn't correct -- you're modifying the layer here, whereas syncToLayer should not touch the layer and just update the GUI to reflect the layer's current state